external/storm-kafka-client/src/main/java/org/apache/storm/kafka/trident/TridentKafkaState.java

```diff
@@ -98,14 +107,19 @@ public void updateState(List<TridentTuple> tuples, TridentCollector collector) {
             }
 
             if(exceptions.size() > 0){
-                String errorMsg = "Could not retrieve result for messages " + tuples + " from topic = " + topic 
-                        + " because of the following exceptions: \n";
-                for (ExecutionException exception : exceptions) {
-                    errorMsg = errorMsg + exception.getMessage() + "\n";
-                }
-                LOG.error(errorMsg);
-                throw new FailedException(errorMsg);
-            }
+		StringBuilder errorMsg = new StringBuilder("Could not retrieve result for messages ").append(tuples)
```

Comment:
Personally I'd rather using StringBuilder to only concat exception messages. For me current implementation is better for readability, and compiler is able to care of this. 
(Please note that those lines are not - and should not be - hot spot, since code block will be executed only when there're some exceptions.)